tests/test.c

```diff
@@ -489,6 +489,42 @@ void check_inputs(uint64_t expected_buttons, float expected_joysticks[4], robot_
     print_pass();
 }
 
+/******************** UDP Device Data Check ******************/
+
+void check_udp_device_exists(DevData* dev_data, int index, uint8_t type, uint64_t uid) {
+    if (index >= dev_data->n_devices) {
+        print_fail();
+        fprintf_delimiter(stderr, "Expected device with type %u uid %llu at index %d but not enough indices", type, uid, index);
+        end_test();
+        exit(1);
+    }
+    Device* device = dev_data->devices[index];
+    if (device->uid != uid || device->type != type) {
+        print_fail();
+        fprintf_delimiter(stderr, "Expected device with type %u uid %llu at index %d but got type %u uid %llu instead",
+                          type, uid, index, device->type, device->uid);
+        end_test();
+        exit(1);
+    }
+    print_pass();
+}
+
+void check_udp_device_param(DevData* dev_data, int dev_idx, char* param_name, param_type_t param_type, param_val_t param_val, uint8_t readonly) {
```

Comment:
Is this function done? What is it used for? Should this be added to the header file?

Reply:
I've finished this function now